src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1653,6 +1653,41 @@ void C2_MacroAssembler::round_double_mode(FloatRegister dst, FloatRegister src,
   bind(done);
 }
 
+// According to Java SE specification, for floating-point signum operations, if
+// on input we have NaN or +/-0.0 value we should return it,
+// otherwise return +/- 1.0 using sign of input.
+// tmp1 - used to store result of fclass operation,
```

Comment:
Seems to me that scratch register `t0` could be used in this function in order to save use of `tmp1`.